circuits/cpp/src/aztec3/circuits/kernel/private/native_private_kernel_circuit_ordering.cpp

```diff
@@ -148,14 +179,29 @@ void match_reads_to_commitments(DummyBuilder& builder,
 //    array_rearrange(new_nullifiers);
 //}
 
-KernelCircuitPublicInputs<NT> native_private_kernel_circuit_ordering(DummyBuilder& builder,
+void apply_commitment_nonces(NT::fr const& first_nullifier,
+                             std::array<NT::fr, MAX_NEW_COMMITMENTS_PER_TX> const& siloed_commitments,
+                             std::array<NT::fr, MAX_NEW_COMMITMENTS_PER_TX>& unique_siloed_commitments)
+{
+    for (size_t c_idx = 0; c_idx < MAX_NEW_COMMITMENTS_PER_TX; c_idx++) {
+        // Apply nonce to all non-zero/non-empty commitments
+        // Nonce is just index in new_commitments array
```

Comment:
This comment's not quite right. A commitment's nonce is: the hash of the first nullifier and the commitment's index in the new commitments array.

Reply:
Fixed